boot_services/src/boot_services.rs

```diff
@@ -1487,4 +1487,21 @@ mod test {
         let status = boot_services.free_pages(0x100000, 10);
         assert!(matches!(status, Ok(())));
     }
+
+    #[test]
+    fn test_allocate_pool() {
+        let boot_services = boot_services!(allocate_pool = efi_allocate_pool);
+
+        extern "efiapi" fn efi_allocate_pool(mem_type: efi::MemoryType, size: usize, mut _buffer: *mut *mut c_void) -> efi::Status {
+            let expected_mem_type: efi::MemoryType = MemoryType::MEMORY_MAPPED_IO.into();
+            assert_eq!(mem_type, expected_mem_type);
+            assert_eq!(size, 10);
+            _buffer = 0x55AA as *mut *mut c_void;
+            efi::Status::SUCCESS
+        }
+
+        let status = boot_services.allocate_pool(MemoryType::MEMORY_MAPPED_IO, 10);
+        let _expected_buffer = 0x55AA as *mut u8;

```

Comment:
Same with `_expected_buffer` - you can name it `expected_buffer` - preceding `_` means "unused" but that isn't the case here.